src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -1542,8 +1553,11 @@ private void doGenCert(String alias, String sigAlgName, InputStream in, PrintStr
             }
         }
 
-        checkWeak(rb.getString("the.issuer"), keyStore.getCertificateChain(alias));
-        checkWeak(rb.getString("the.generated.certificate"), cert);
+        checkWeakConstraint(rb.getString("the.issuer"),
+                keyStore.getCertificateChain(alias));
+        cpcp = new CertPathConstraintsParameters(cert, null, null, null);
```

Comment:
You could also specify the variant parameter if the certificate contains an EKU extension. The mapping from EKU to variant would be:

- anyExtendedKeyUsage -> VAR_GENERIC
- serverAuth -> VAR_TLS_SERVER
- clientAuth -> VAR_TLS_CLIENT
- codeSigning -> VAR_CODE_SIGNING
- emailProtection -> none defined, so just use null or VAR_GENERIC
- timeStamping -> VAR_TSA_SERVER
- OCSPSigning -> none defined, so just use null or VAR_GENERIC

Reply:
Fixed.